torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -199,21 +200,48 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                     "FSDP requires the DP and TP mesh to have the same parent mesh but got: \n"
                     f"DP's global mesh: {dp_global_mesh}\nTP's global mesh: {tp_global_mesh}"
                 )
-            self._global_mesh = dp_global_mesh
             if len(self._tp_spec.placements) != 1:
                 raise NotImplementedError(
                     f"FSDP only supports 1D TP, not {self._tp_spec.placements}"
                 )
             global_placements: List[Placement] = [Replicate(), Replicate()]
             global_dp_mesh_dim = _mesh_resources.get_parent_mesh_dim(dp_mesh)
             global_tp_mesh_dim = _mesh_resources.get_parent_mesh_dim(tp_mesh)
+
+            def infer_spmd_mesh(global_mesh, global_dp_mesh_dim, global_tp_mesh_dim):
+                from torch.distributed.device_mesh import _mesh_resources, DeviceMesh
+                pp_submesh_tensors = global_mesh.mesh.unbind(0)
+                spmd_mesh_dim_names = (
+                    global_mesh.mesh_dim_names[global_dp_mesh_dim],
+                    global_mesh.mesh_dim_names[global_tp_mesh_dim],
+                )
+                for spmd_mesh_tensor in pp_submesh_tensors:
```

Comment:
i couldn't decide if `spmd_mesh_tensor` represents 'some tensor we created on the old mesh' or 'a tensor that represents the ranks that are in a submesh'

Reply:
ok i think i see whats going on.  maybe global mesh starts like 
```
tensor([[[0, 1],
         [2, 3]],

        [[4, 5],
         [6, 7]]])
```

and unbind(0) deletes the outer dim and returns two submesh tensors that looks like 
```
tensor([[0, 1],
        [2, 3]])

tensor([[4, 5],
        [6, 7]])
```

then you check which one your global rank is in and thats your tensor

Reply:
yes!

Reply:
but this seems like a hard way to accomplish the goal.  what happens if you just slice out a submesh instead?

could we support something like 
```
world_mesh[("dp", "tp")]
[rank0]:(Pdb) [rank0]:*** KeyError: ("Mesh dimension '('dp', 'tp')' does not exist.", "Available mesh dimensions are: mesh_dim_names=('pp', 'dp', 'tp')")
```

Reply:
Yeah I think that would be better to live in FSDP code, what I implemented here is essentially the submesh slicing implementation logic. cc @wz337

Reply:
> Yeah I think that would be better to live in FSDP code,

you mean, better to live in device-mesh code?

Reply:
oh, ok. got it. yea, agreed